local-cli/automated-vs/solutions/windowsSolution.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * Some descendant of this class must have:
+ * getWindowsSDKFolderPath
+ */
+
+'use-strict';
+
+const execSync = require('child_process').execSync;
+const shell = require('shelljs');
+const path = require('path');
+const Version = require('../utils/version');
+const VSSolution = require('./vsSolution');
+
+module.exports = class WindowsSolution extends VSSolution {
+    constructor(slnFile, root, minWinSDK) {
+        super(slnFile, root);
+
+        this.minWinSDK = minWinSDK;
+    }
+
+    //build
+    checkAndGetRequirements() {
+        super.checkAndGetRequirements();
+        this.checkWinSDK(this.minWinSDK);
+    }
+
+    checkWinSDK(version) {
+        const minVersion = Version.fromString(version);
+        const allVersions = this.getAllAvailableWinSDKVersions();
+
+        if (!allVersions.some(v => v.gte(minVersion))) {
+            throw new Error('Must have a minimum Windows SDK version' + minVersion + 'installed');
+          }
+    }
+
+    getAllAvailableWinSDKVersions() {
+        let results = this.checkForWinSDKsInProgramFiles();
+        if (results.length <= 0) {
+            results = this.checkForWinSDKsInReg();
+        }
+
+        return results;
+    }
+
+    checkForWinSDKsInProgramFiles() {
+        const installedSdks = [];
+
+        const programFilesFolder = process.env['ProgramFiles(x86)'] || process.env.ProgramFiles;
+
+        if (!programFilesFolder) {
+            return installedSdks;
+        }
+
+        var folderPath = this.getWindowsSDKFolderPath(programFilesFolder);
+
+        if (!shell.test('-e', folderPath)) { //if folder path doesn't exist
+            return installedSdks;
+        }
+
+        shell.ls(folderPath)
+            .filter(uapDir => shell.test('-d', path.join(folderPath, uapDir))) //if path is a directory
+            .map(Version.tryParse)
+            .forEach(version => version && installedSdks.push(version));
+
+        return installedSdks;
+    }
+
+    checkForWinSDKsInReg() { //keeping the registry check for generalizability but the user will have to specify a less specific version of win sdk - only gets 10 or 8.1
```

Comment:
if automated-vs has to default to checking the registry, the build will fail with the minWinSDK of 10.0.14393.0, because the registry only detects if the win sdk is version 10.0.0.0.

I kept the registry check so the user could specifiy a less specific version if they have to depend on the registry.

Another option is to make a round-down function in version.js that takes a specific version number and rounds down to just the major number.